modules/core/src/test/java/org/apache/ignite/internal/processors/performancestatistics/CheckpointTest.java

```diff
@@ -0,0 +1,228 @@
+
+        AtomicInteger cnt = new AtomicInteger();
+
+        stopCollectStatisticsAndRead(new TestHandler() {
+            @Override public void checkpoint(
+                UUID nodeId,
+                long beforeLockDuration,
+                long lockWaitDuration,
+                long listenersExecDuration,
+                long markDuration,
+                long lockHoldDuration,
+                long pagesWriteDuration,
+                long fsyncDuration,
+                long walCpRecordFsyncDuration,
+                long writeCpEntryDuration,
+                long splitAndSortCpPagesDuration,
+                long totalDuration,
+                long cpStartTime,
+                int pagesSize,
+                int dataPagesWritten,
+                int cowPagesWritten
+            ) {
+                assertEquals(srv.localNode().id(), nodeId);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointBeforeLockDuration").value(),
+                    beforeLockDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointLockWaitDuration").value(),
+                    lockWaitDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointListenersExecuteDuration").value(),
+                    listenersExecDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointMarkDuration").value(), markDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointLockHoldDuration").value(),
+                    lockHoldDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointPagesWriteDuration").value(),
+                    pagesWriteDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointFsyncDuration").value(), fsyncDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointWalRecordFsyncDuration").value(),
+                    walCpRecordFsyncDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointWriteEntryDuration").value(),
+                    writeCpEntryDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointSplitAndSortPagesDuration").value(),
+                    splitAndSortCpPagesDuration);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointDuration").value(), totalDuration);
+                assertEquals(lastStart.value(), cpStartTime);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointTotalPagesNumber").value(), pagesSize);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointDataPagesNumber").value(),
+                    dataPagesWritten);
+                assertEquals(mreg.<LongMetric>findMetric("LastCheckpointCopiedOnWritePagesNumber").value(),
+                    cowPagesWritten);
+
+                cnt.incrementAndGet();
+            }
+        });
+
+        assertEquals(1, cnt.get());
```

Comment:
The test became flaky. This happens because the statistics stop immediately after the metric is updated, without waiting for the statistics to be written.

It fails locally 1/100:
```
[2021-04-14 09:25:08,548][ERROR][main][root] Test failed [test=CheckpointTest#testCheckpoint, duration=725]
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.apache.ignite.testframework.junits.JUnitAssertAware.assertEquals(JUnitAssertAware.java:94)
	at org.apache.ignite.internal.processors.performancestatistics.CheckpointTest.testCheckpoint(CheckpointTest.java:165)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.apache.ignite.testframework.junits.GridAbstractTest$7.run(GridAbstractTest.java:2391)
	at java.lang.Thread.run(Thread.java:748)
```